src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/RazorSemanticTokensInfoService.cs

```diff
@@ -91,19 +100,72 @@ internal class RazorSemanticTokensInfoService : IRazorSemanticTokensInfoService
             return null;
         }
 
+        // Because SemanticToken data is generated relative to the previous token it must be in order.
+        // We have a guarantee of order within any given language server, but when we translate from the ranges
+        // in the C# document to ranges into the Razor document we lose that guarantee.
+        // Having converted them to SemanticRange objects, we can simply do the final round of a merge sort.
         var newList = new List<SemanticRange>(ranges1.Count + ranges2.Count);
-        newList.AddRange(ranges1);
-        newList.AddRange(ranges2);
+        var idxRange1 = 0;
```

Comment:
nit: I think we usually spell things out. This reminds me of hungarian notation.

Reply:
Agree, I had to think for a bit on what `idx` stood for (I'm assuming `index`?)

Reply:
Done